The Straits Times.js

```diff
@@ -0,0 +1,286 @@
+		return 'webpage';
+	}
+	var pageClass = ZU.xpathText(doc, '//meta[@name="cXenseParse:pageclass"]/@content');
+	if (pageClass === 'article') {
+		return 'newspaperArticle';
+	}
+	if (pageClass === 'frontpage') {
+		return 'multiple';
+	}
+	
+	return false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) === "multiple") {
+		getMultipleItems(doc, url);
+	} else {
+		scrape(doc, url, detectWeb(doc, url));
+	}
+}
+
+function scrape(doc, url, type) {
+	var newItem = new Zotero.Item(type || 'newspaperArticle');
+	newItem.ISSN = '0585-3923';
+	newItem.url = url;
+	newItem.publicationTitle = 'The Straits Times';
+	newItem.title = ZU.xpathText(doc, '//meta[@name="dcterms.title"]/@content');
+	newItem.abstractNote = ZU.xpathText(doc, '//meta[@name="dcterms.description"]/@content') ? ZU.xpathText(doc, '//meta[@name="dcterms.description"]/@content').replace('Read more at straitstimes.com.', '').trim() : '';
+	newItem.date = ZU.xpathText(doc, '//meta[@name="dcterms.date"]/@content');
+	newItem.place = 'Singapore';
+	newItem.language = 'en';
+	var authors = ZU.xpath(doc, '//meta[@property="article:author"]');
+	for (var i = 0; i < authors.length; i++) {
+		insertCreator(authors[i].getAttribute('content'), newItem);
+	}
+	newItem.attachments = [{
+		document: doc,
+		title: "The Straits Times Snapshot",
+	}];
+	if (doc.evaluate('//div[@class="paid-premium st-flag-1"]', doc, null, XPathResult.ANY_TYPE, null).iterateNext()) {
+		newItem.extra = "(Subscription only)";
+	}
+	newItem.complete();
+}
+
+function getMultipleItems(doc, url) {
+	var items = [];
+	var rows;
+	if (url.includes('/search?') && url.includes('searchKey')) {
+		rows = ZU.xpath(doc, '//div[@class="queryly_item_row"]');
+		if (rows.length) {
+			for (var i = 0; i < rows.length; i++) {
+				var searchItem = rows[i];
+				var searchItemUrl = searchItem.querySelector('a') ? searchItem.querySelector('a').href : '';
+				items.push(searchItemUrl);
+			}
+		}
+	} else {
+		rows = ZU.xpath(doc, '//a[@class="block-link"]|//span[@class="story-headline"]/a');
+		if (rows.length) {
+			for (var k = 0; k < rows.length; k++) {
+				var headlineItem = rows[k];
+				var headlineItemUrl = headlineItem.href ? headlineItem.href : '';
+				items.push(headlineItemUrl);
+			}
+		}
+	}
+	if (!!items && items.length) {
+		items = items.filter(function(item) {
+			return (!!item.match(/^https:\/\/www\.straitstimes.com/));
+		});
+		if (items.length) {
+			ZU.processDocuments(items, scrape);
+		}
+	}
+}
+
+function insertCreator(authorName, newItem) {
+	// to account for mostly Chinese names in formats of: <last> <first>, <first (typically English)> <last> <first (typically Chinese)>
+	// list derived from searching in Google with the follow search pattern: authors site:straitstimes.com/authors
+	var authorList = {
+		'Alison de Souza': { first: 'Alison', last: 'de Souza' },
+		'Aw Cheng Wei': { first: 'Cheng Wei', last: 'Aw' },
+		'Benjamin Kang Lim': { first: 'Kang', last: 'Benjamin, Lim' },
```

Comment:
His English given name is Benjamin, his Chinese given name is Lim, and his surname is Kang, right? In that case, we want to store this as `{ first: 'Benjamin (Lim)', last: 'Kang' }` (and similarly for other names in the list). I'm not sure how citation processors will handle a name with a comma in it, but it's less standard than parentheses, and we definitely shouldn't reverse the order and put first names in `last`.

Reply:
Nope. English given name: Benjamin, Chinese given name is Kang, Surname is Lim. It took awhile for me to figure that out, but the hint is in the email of the journalist. Now looking at this, it might have been a mistake here. I should have put 'Benjamin, Kang' in the first parameter, and Lim in the last parameter. I think I will just simply put his first as 'Benjamin Kang'.